plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/parser/exchange/binance/v4/BinanceSortedGroupV4.java

```diff
@@ -0,0 +1,281 @@
+        int withdrawNum = rowWithdrawal.size();
+        int feeNum = rowFees.size();
+
+        if (buySellNum > 0) {
+            validateBuySell();
+            createBuySellTxs();
+        }
+
+        if (depositNum > 0 || withdrawNum > 0) {
+            createDepositWithdrawalTxs();
+        }
+
+        if (feeNum > 0) {
+            validateFee();
+            addFeeToTxs();
+        }
+        createdTransactions.stream().forEach(r -> {
+            r.setInTransaction(true);
+        });
+    }
+
+    private void addFeeToTxs() {
+        try {
+            var txs = createdTransactions.get(0);
+            for (BinanceBeanV4 fee : rowFees) {
+                fee.setInTransaction(true);
+                var bean = new BinanceBeanV4();
+                bean.setType(TransactionType.FEE);
+                bean.setFee(fee.getChange());
+                bean.setFeeCurrency(fee.getCoin());
+                bean.setDate(fee.getDate());
+                bean.setOperation(fee.getOperation());
+                bean.setRowId(fee.getRowId());
+                bean.setMessage(fee.getMessage());
+                txs.feeTransactions.add(bean);
+                txs.usedIds.addAll(fee.usedIds);
+                txs.setMergedWithAnotherGroup(fee.isMergedWithAnotherGroup());
+            }
+        } catch (Exception e) {
+            throw new DataIgnoredException("Fee not assigned to transaction");
+        }
+    }
+
+    private String parseIds(List<Integer> ids) {
+        String s = "";
+        for (int id : ids) {
+            s = s + " " + id + "; ";
+        }
+        return s;
+    }
+
+    private void createDepositWithdrawalTxs() {
+        List<BinanceBeanV4> mergeDepositWithdrawal = new ArrayList<>();
+        mergeDepositWithdrawal.addAll(rowDeposit);
+        mergeDepositWithdrawal.addAll(rowWithdrawal);
+        var idsList = mergeDepositWithdrawal.stream().map(BinanceBeanV4::getRowId).collect(Collectors.toList());
+        var ids = parseIds(idsList);
+        for (BinanceBeanV4 row : mergeDepositWithdrawal) {
+            var txs = new BinanceBeanV4();
+            txs.setRowNumber(row.getDate().getEpochSecond());
+            String[] strings = {"Row id " + ids + " " + row.getOperation()};
+            txs.setRowValues(strings);
+            txs.usedIds.addAll(row.usedIds);
+            txs.setAmountBase(row.getChange().abs());
+            txs.setMarketBase(row.getCoin());
+            txs.setDate(row.getDate());
+            txs.setMergedWithAnotherGroup(row.isMergedWithAnotherGroup());
+            if (row.getChange().compareTo(BigDecimal.ZERO) > 0) {
+                txs.setType(TransactionType.DEPOSIT);
+            } else {
+                txs.setType(TransactionType.WITHDRAWAL);
+            }
+            createdTransactions.add(txs);
+        }
+    }
+
+    private void createBuySellTxs() {
+        var stRow = rowBuySellRelated.get(0);
+        var ndRow = rowBuySellRelated.get(1);
```

Comment:
what if rowBuySellRelated is empty?

Reply:
It cannot be empty, because before this method is validation
        if (rowBuySellRelated.size() != 2) {
            throw new DataIgnoredException("Wrong number of currencies");
        }